pyzo/codeeditor/extensions/appearance.py

```diff
@@ -166,36 +222,36 @@ def _findMatchingBracket(self, char, cursor, doc):
         """
         if char in ')]}':
             direction = -1
+            stacking = ')]}'
+            unstacking = '([{'
         elif char in '([{':
             direction = 1
+            stacking = '([{'
+            unstacking = ')]}'
         else:
             raise ValueError('invalid bracket character: ' + char)
             
         other_char = self._matchingBrackets[char]
-        fulltext = doc.toPlainText()
-        pos = cursor.position() - 1
-        num_match = 0
-        while True:
-            if pos > len(fulltext)-1 or pos < 0:
-                return None
-            
-            if fulltext[pos] == char:
-                num_match += 1
-            elif fulltext[pos] == other_char:
-                num_match -= 1
-            
-            pos += direction
-            
-            if num_match == 0:
+        stacked_paren = [char] # using a Python list as a stack
+        # stack not empty because the _ParenIterator will not give back
+        # the parenthesis we're matching
+        for (token, bloc_pos) in _ParenIterator(cursor, direction) :
+            paren = token._style
+            if paren in stacking :
+                stacked_paren.append(paren)
+            elif paren in unstacking :
+                if stacked_paren == [] or self._matchingBrackets[stacked_paren[-1]] != paren :
```

Comment:
IIUC the second part of this `if` represents a mismatch. Would it be good (and feasible) to highlight (in red) both braces that make up the mismatch?